RiotSharp/TournamentRiotApi.cs

```diff
@@ -0,0 +1,509 @@
+        /// <param name="name">The optional name of the tournament.</param>
+        /// <returns></returns>
+        public async Task<Tournament> CreateTournamentAsync(int providerId, string name)
+        {
+            var body = new Dictionary<string, object> {{"name", name}, {"providerId", providerId}};
+            var json =
+                await
+                    requester.CreatePostRequestAsync(TournamentRootUrl + CreateTournamentUrl, Region.global,
+                        JsonConvert.SerializeObject(body));
+
+            // json is an int directly
+            var obj = new Tournament {Id = int.Parse(json)};
+
+            return await Task.Factory.StartNew(() => obj);
+        }
+
+        /// <summary>
+        ///     Create a tournament code for the given tournament id.
+        /// </summary>
+        /// <param name="tournamentId">The tournament ID</param>
+        /// <param name="teamSize">The team size of the game. Valid values are 1-5.</param>
+        /// <param name="allowedSummonerIds">
+        ///     Optional list of participants in order to validate the players eligible to join the
+        ///     lobby. NOTE: We currently do not enforce participants at the team level, but rather the aggregate of teamOne and
+        ///     teamTwo. We may add the ability to enforce at the team level in the future.
+        /// </param>
+        /// <param name="spectatorType">The spectator type of the game.</param>
+        /// <param name="pickType">The pick type of the game.</param>
+        /// <param name="mapType">The map type of the game.</param>
+        /// <param name="metadata">
+        ///     Optional string that may contain any data in any format, if specified at all. Used to denote any
+        ///     custom information about the game.
+        /// </param>
+        /// <returns>The tournament code in string format.</returns>
+        public string CreateTournamentCode(int tournamentId, int teamSize, List<long> allowedSummonerIds,
+            TournamentSpectatorType spectatorType, TournamentPickType pickType, TournamentMapType mapType,
+            string metadata)
+        {
+            var body = new Dictionary<string, object>
+            {
+                {"teamSize", teamSize},
+                {"allowedSummonerIds", new Dictionary<string, object> {{"participants", allowedSummonerIds}}},
+                {"spectatorType", spectatorType},
+                {"pickType", pickType},
+                {"mapType", mapType},
+                {"metadata", metadata}
+            };
+            var json = requester.CreatePostRequest(TournamentRootUrl + CreateCodeUrl, Region.global,
+                JsonConvert.SerializeObject(body),
+                new List<string> {string.Format("tournamentId={0}", tournamentId), string.Format("count={0}", 1)});
+
+            // json is a list of strings
+            var obj = JsonConvert.DeserializeObject<List<string>>(json);
```

Comment:
could you rename `obj` to `tournamentCodes` or something similar to make it a bit more explicit?